pyanaconda/network.py

```diff
@@ -1151,14 +1242,17 @@ def write_hostname(rootpath, ksdata, overwrite=False):
 
 def disableIPV6(rootpath):
     cfgfile = os.path.normpath(rootpath + ipv6ConfFile)
-    if ('noipv6' in flags.cmdline
-        and all(nm.nm_device_setting_value(dev, "ipv6", "method") == "ignore"
-                for dev in nm.nm_devices() if nm.nm_device_type_is_ethernet(dev))):
-        log.info('Disabling ipv6 on target system')
-        with open(cfgfile, "a") as f:
-            f.write("# Anaconda disabling ipv6 (noipv6 option)\n")
-            f.write("net.ipv6.conf.all.disable_ipv6=1\n")
-            f.write("net.ipv6.conf.default.disable_ipv6=1\n")
+    try:
+        if ('noipv6' in flags.cmdline
+            and all(nm.nm_device_setting_value(dev, "ipv6", "method") == "ignore"
+                    for dev in nm.nm_devices() if nm.nm_device_type_is_ethernet(dev))):
```

Comment:
It is much clearer to me what it should do now. I'm fine with this.

I just want to point out the difference:

Original:
```
 if ('noipv6' in flags.cmdline
            and all(nm.nm_device_setting_value(dev, "ipv6", "method") == "ignore"
                    for dev in nm.nm_devices() if nm.nm_device_type_is_ethernet(dev))):
```

New solution:
```
    if 'noipv6' in flags.cmdline:
        for devname in nm.nm_devices():
            if nm.nm_device_type_is_ethernet(devname):
                try:
                    ipv6_method = nm.nm_device_setting_value(devname, "ipv6", "method")
                except nm.MultipleSettingsFoundError as e:
                    log.debug("network: %s when getting ipv6 method of %s", e, devname)
                    ipv6_method = None
                if ipv6_method != "ignore":
                    return
    log.info('network: disabling ipv6 on target system')
    cfgfile = os.path.normpath(rootpath + ipv6ConfFile)
    with open(cfgfile, "a") as f:
        f.write("# Anaconda disabling ipv6 (noipv6 option)\n")
        f.write("net.ipv6.conf.all.disable_ipv6=1\n")
        f.write("net.ipv6.conf.default.disable_ipv6=1\n")
```

My proposed solution:
```
if 'noipv6' in flags.cmdline:
    if self._find_all_ethernet_devices_with_ipv6_ignored():
        return
   cfgfile = os.path.normpath(rootpath + ipv6ConfFile)
    with open(cfgfile, "a") as f:
        f.write("# Anaconda disabling ipv6 (noipv6 option)\n")
        f.write("net.ipv6.conf.all.disable_ipv6=1\n")
        f.write("net.ipv6.conf.default.disable_ipv6=1\n")
```

Line numbers will be almost the same (+2 empty lines and one function declaration) the only difference is readability for others. Of course this is just my point of view.